Sources/Datadog/RUM/RUMMonitor/Scopes/RUMSessionScope.swift

```diff
@@ -135,51 +158,63 @@ internal class RUMSessionScope: RUMScope, RUMContextProvider {
-            let handlingRule = RUMOffViewEventsHandlingRule(
-                sessionState: state,
-                isAppInForeground: context.applicationStateHistory.currentSnapshot.state.isRunningInForeground,
-                isBETEnabled: backgroundEventTrackingEnabled
-            )
+        var deactivating = false
+        if isActive {
+            if command is RUMStopSessionCommand {
+                isActive = false
+                deactivating = true
+            } else if let startApplicationCommand = command as? RUMApplicationStartCommand {
+                startApplicationLaunchView(on: startApplicationCommand, context: context, writer: writer)
+            } else if let startViewCommand = command as? RUMStartViewCommand {
+                // Start view scope explicitly on receiving "start view" command
+                startView(on: startViewCommand, context: context)
+            } else if !hasActiveView {
+                // Otherwise, if there is no active view scope, consider starting artificial scope for handling this command
+                let handlingRule = RUMOffViewEventsHandlingRule(
+                    sessionState: state,
+                    isAppInForeground: context.applicationStateHistory.currentSnapshot.state.isRunningInForeground,
+                    isBETEnabled: backgroundEventTrackingEnabled
+                )
 
-            switch handlingRule {
-            case .handleInBackgroundView where command.canStartBackgroundView:
-                startBackgroundView(on: command, context: context)
-            default:
-                if !(command is RUMKeepSessionAliveCommand) { // it is expected to receive 'keep alive' while no active view (when tracking WebView events)
-                    // As no view scope will handle this command, warn the user on dropping it.
-                    DD.logger.warn(
+                switch handlingRule {
+                case .handleInBackgroundView where command.canStartBackgroundView:
+                    startBackgroundView(on: command, context: context)
+                default:
+                    if !(command is RUMKeepSessionAliveCommand) { // it is expected to receive 'keep alive' while no active view (when tracking WebView events)
+                        // As no view scope will handle this command, warn the user on dropping it.
+                        DD.logger.warn(
                         """
                         \(String(describing: command)) was detected, but no view is active. To track views automatically, try calling the
                         DatadogConfiguration.Builder.trackUIKitRUMViews() method. You can also track views manually using
                         the RumMonitor.startView() and RumMonitor.stopView() methods.
                         """
-                    )
+                        )
+                    }
                 }
             }
         }
```

Comment:
The conditions and state management are difficult to follow and error prone IMO. It might be helpful to split this into multiple methods, would that be possible?

Reply:
I moved the "no active view" portion to its own function. Let me know if that helps or it you think it needs more.

Reply:
Perfect, thank you!